src/main/java/com/google/enterprise/secmgr/config/JsonConfig.java

```diff
@@ -86,7 +86,8 @@ protected SecurityManagerConfig readConfigInternal(Reader reader)
     SecurityManagerConfig config;
     try {
       config = ConfigSingleton.getGson().fromJson(je, SecurityManagerConfig.class);
-    } catch (JsonParseException e) {
+    } catch (Exception e) {
+      // Replicating catch-all workaround from an internal G-patch
```

Comment:
what kind of exceptions do we get here?

Reply:
the internal G patch of this gson library catches all exceptions (Exception) and re-throws them as JsonParseExceptions. Since we don't have this patch in place with OSS version, I replicated the catch-all branch here where we use Gson.

Our tests are effectively failing with an InvalidArgumentException because of missing this internal G patch, however, it can be anything else and thought better to handle all exceptions the same way, as it was a forced JsonParseException in any cases..